### PR TITLE
Revert "accept variable number of key, val in log.WithKeyValue"

### PR DESCRIPTION
**Pull request by alovak**
Reverts moov-io/base#92

Motivation for this revert is to reduce the chance of passing wrong amount of arguments to the method and getting broken logs. As example,

```go
log.WithKeyValue("one", "1", "two", "2", "three", "3")
```
Say you miss the "1" in that line and now your indexes are all off. You'd end up with "one=two" "2=three" in logs. 

Versus with fixed number of arguments (this PR) you are constrainted by the compiler to not mess it up:

```go
log.WithKeyValue("one", "1").WithKeyValue("two", "2")
```
Also, if you try to add 3 like this:

```go
log.WithKeyValue("three", ) 
```
it would compiler error.

Based on discussion with @InfernoJJ

### CODE REVIEW

**Comment by adamdecaf** [comment]
cc @InfernoJJ

File: log/logger.go [hunk 1]
```diff
@@ -13,7 +13,7 @@ import (
 type Logger interface {
 	With(ctxs ...Context) Logger
 	WithMap(mapCtx map[string]string) Logger
-	WithKeyValue(keyvals ...string) Logger
+	WithKeyValue(key, value string) Logger
```
Comment (vxio) [inline log/logger.go:16]: Nitpick:
Could we shorten this to `WithKV`?

Comment (alovak) [reply log/logger.go:16]: I would make it `With` and `WithContext` depending on the frequency of usage :)

From my recent experience I used `WithKeyValue` a lot and zero times `With`.

Comment (alovak) [reply log/logger.go:16]: @InfernoJJ what's your experience?

Comment (vxio) [reply log/logger.go:16]: Agreed, looking at our `identity` service, `WithKeyValue()` is called a lot more frequently than `With(ctx ...Context)`. 
Renaming:
- `WithKeyValue` -> `With`
- `With` -> `WithContexts` 
would be great.

Comment (alovak) [reply log/logger.go:16]: @InfernoJJ thoughts?

Comment (alovak) [reply log/logger.go:16]: @vxio let's keep it as it. we can rename it pretty quickly if all projects together. Right now it may be a little bit complicated.

**Comment by alovak** [comment]
@InfernoJJ please, approve :) It will take no time for you as I'm just reverting the code.

**Review by InfernoJJ — APPROVED** [review]

**Review by vxio — APPROVED** [review]
Dang, I liked this change.

**Comment by alovak** [comment]
@vxio we will add:

```go
logger.With(log.Fields{
  "one": "1",
  "two": "2"
})
```

will it compensate your upset?
